File: 051/7.py

class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age


    def displayDetails(self):
        print(f"ชื่อ: {self.name}")
        print(f"อายุ: {self.age}")


class Teacher(Person):
    def __init__(self, name, age, subject):
        super().__init__(name, age)
        self.subject = subject


    def displayDetails(self):
        super().displayDetails()
        print(f"วิชาที่สอน: {self.subject}")


class Student(Person):
    def __init__(self, name, age, grade):
        super().__init__(name, age)
        self.grade = grade


    def displayDetails(self):
        super().displayDetails()
        print(f"เกรด: {self.grade}")


teacher = Teacher("อาจารย์สมชาย", 40, "คอมพิวเตอร์")
teacher.displayDetails()

print("\n")


student = Student("ณทชัย", 16, "มัธยมศึกษาปีที่ 6")
student.displayDetails()

File: 051/7.js

class Person {
    constructor(name, age) {
      this.name = name;
      this.age = age;
    }
  
 
    displayDetails() {
      console.log(`ชื่อ: ${this.name}`);
      console.log(`อายุ: ${this.age}`);
    }
  }
  

  class Teacher extends Person {
    constructor(name, age, subject) {
      super(name, age);
      this.subject = subject;
    }
  

    displayDetails() {
      super.displayDetails();
      console.log(`วิชาที่สอน: ${this.subject}`);
    }
  }
  
 
  class Student extends Person {
    constructor(name, age, grade) {
      super(name, age);
      this.grade = grade;
    }
  
 
    displayDetails() {
      super.displayDetails();
      console.log(`เกรด: ${this.grade}`);
    }
  }

  const teacher = new Teacher("อาจารย์ไกร", 40, "คอมพิวเตอร์");
  teacher.displayDetails();
  
  console.log("\n");
  

  const student = new Student("ณทชัย", 16, "มัธยมศึกษาปีที่ 6");
  student.displayDetails();
  